build/purger/purger.py

```diff
@@ -263,17 +220,31 @@ def update_user_activity(
     return network_to_due_users
 
 
+def get_discovery_room(api: GMatrixHttpApi, network_value: int) -> Optional[RoomInfo]:
+
+    server = urlparse(api.base_url).netloc
+    discovery_room_alias = make_room_alias(ChainID(network_value), DISCOVERY_DEFAULT_ROOM)
+    local_room_alias = f"#{discovery_room_alias}:{server}"
+
+    try:
+        room_id = api.get_room_id(local_room_alias)
+        return RoomInfo(room_id, discovery_room_alias, server)
+    except MatrixError as ex:
+        click.secho(f"Could not find room {discovery_room_alias} with error {ex}")
+
+    return None
+
+
 def _fetch_new_members_for_network(
     api: GMatrixHttpApi, user_activity: Dict[str, int], discovery_room: RoomInfo, current_time: int
 ) -> None:
     try:
-        response = api.get_room_members(discovery_room.room_id)
+        response = api._send("GET", f"/_synapse/admin/v1/rooms/{discovery_room.room_id}/members")
+        server_name = urlparse(api.base_url).netloc
         room_members = [
-            event["state_key"]
-            for event in response["chunk"]
-            if event["content"]["membership"] == "join"
-            and event["state_key"].split(":")[1] == urlparse(api.base_url).netloc
-            and not event["state_key"].find("admin")
+            member
+            for member in response["members"]
+            if member.split(":")[1] == server_name and not member.startswith("@admin")
```

Comment:
It shouldn't be necessary to change this here but just FYI:

In cases like this where a string is expected to contain a separator, using [`.partition()`](https://docs.python.org/3.8/library/stdtypes.html#str.partition) is usually preferable since it will always return a tuple with three values. This means a missing separator will not cause an IndexError (unlike with `.split()`)